sdks/python/apache_beam/pipeline.py

```diff
@@ -564,3 +685,43 @@ def from_runner_api(proto, context):
           pc.tag = tag
     result.update_input_refcounts()
     return result
+
+
+class PTransformMatcher(object):
```

Comment:
FWIW, Pythonic way to do this would be to return a callable, not have a hierarchy of PTransformMatchers. (These callables could be put in a library for re-use if need be.)